test/orderbook_service_test.ts

```diff
@@ -0,0 +1,226 @@
+import { DummyERC20TokenContract, WETH9Contract } from '@0x/contracts-erc20';
+import { constants, expect, OrderFactory } from '@0x/contracts-test-utils';
+import { BlockchainLifecycle, web3Factory, Web3ProviderEngine } from '@0x/dev-utils';
+import { OrderEventEndState } from '@0x/mesh-rpc-client';
+import { assetDataUtils, Order, orderHashUtils } from '@0x/order-utils';
+import { BigNumber } from '@0x/utils';
+import { Web3Wrapper } from '@0x/web3-wrapper';
+import 'mocha';
+
+import * as config from '../src/config';
+import { DEFAULT_PAGE, DEFAULT_PER_PAGE } from '../src/constants';
+import { getDBConnectionAsync } from '../src/db_connection';
+import { SignedOrderEntity } from '../src/entities';
+import { PersistentSignedOrderEntity } from '../src/entities/PersistentSignedOrderEntity';
+import { OrderBookService } from '../src/services/orderbook_service';
+import { APIOrderWithMetaData } from '../src/types';
+import { MeshClient } from '../src/utils/mesh_client';
+import { orderUtils } from '../src/utils/order_utils';
+
+import {
+    setupDependenciesAsync,
+    setupMeshAsync,
+    teardownDependenciesAsync,
+    teardownMeshAsync,
+} from './utils/deployment';
+import { DEFAULT_MAKER_ASSET_AMOUNT, MeshTestUtils } from './utils/mesh_test_utils';
+
+const SUITE_NAME = 'OrderBook Service tests';
+
+const EMPTY_PAGINATED_RESPONSE = {
+    perPage: DEFAULT_PER_PAGE,
+    page: DEFAULT_PAGE,
+    total: 0,
+    records: [],
+};
+
+const TOMORROW = new BigNumber(Date.now() + 24 * 3600); // tslint:disable-line:custom-no-magic-numbers
+
+async function saveSignedOrderAsync(apiOrder: APIOrderWithMetaData): Promise<void> {
+    await (await getDBConnectionAsync()).manager.save(orderUtils.serializeOrder(apiOrder));
+}
+
+async function savePersistentOrderAsync(apiOrder: APIOrderWithMetaData): Promise<void> {
+    await (await getDBConnectionAsync()).manager.save(orderUtils.serializePersistentOrder(apiOrder));
+}
+
+async function deleteSignedOrderAsync(orderHash: string): Promise<void> {
+    await (await getDBConnectionAsync()).manager.delete(SignedOrderEntity, orderHash);
+}
+
+async function deletePersistentOrderAsync(orderHash: string): Promise<void> {
+    await (await getDBConnectionAsync()).manager.delete(PersistentSignedOrderEntity, orderHash);
+}
+
+async function newAPIOrderAsync(
+    orderFactory: OrderFactory,
+    params: Partial<Order>,
+    remainingFillableAssetAmount?: BigNumber,
+): Promise<APIOrderWithMetaData> {
+    const order = await orderFactory.newSignedOrderAsync({
+        expirationTimeSeconds: TOMORROW,
+        ...params,
+    });
+    const apiOrder: APIOrderWithMetaData = {
+        order,
+        metaData: {
+            orderHash: orderHashUtils.getOrderHash(order),
+            remainingFillableTakerAssetAmount: remainingFillableAssetAmount || order.takerAssetAmount,
+            state: OrderEventEndState.Added,
+        },
+    };
+    return apiOrder;
+}
+
+describe.only(SUITE_NAME, () => {
```

Comment:
Just a reminder to remove the `only` before merging 👍 
```suggestion
describe(SUITE_NAME, () => {
```

Reply:
thanks, good catch!